compiler/rustc_error_codes/src/error_codes/E0161.md

```diff
@@ -40,3 +40,22 @@ fn main() {
     // ok!
 }
 ```
+
+If you absolutely must use a value by moving in a trait, you can use the `Box<Self>`
+pattern. You can then move the value out of the `Box` by dereferencing. For example:
+
+```
+trait Bar {
+    fn f(self: Box<Self>);
+}
+
+impl Bar for String {
+    fn f(self: Box<Self>) {}
+}
+
+fn main() {
+    let b: Box<dyn Bar> = Box::new(0i32);
```

Comment:
Should this be a String inside the box? Your example does not compile.

Reply:
Yes it should be, thank you for that!